Components/AppTabNavigator/StocksTab.js

```diff
@@ -23,16 +23,76 @@ class StocksTab extends Component {
         companyNames: [],
         companySymbols: [],
       },
+      favoritedCompanies: {
+        companyNames: [],
+        companySymbols: [],
+      },
     };
   }
 
   componentDidMount() {
-    //NetInfo.isConnected.fetch().then(isConnected => {
     if (true) {
       this.getStockCardData();
+      this.readFavorites();
     }
+
+    this.willFocusSubscription = this.props.navigation.addListener(
+      'focus',
+      () => {
+        this.readFavorites();
+        console.log("BACK BUTTON");
+      }
+    );
+
+  }
+
+  componentWillUnmount() {
+    this.willFocusSubscription.remove();
   }
 
+  componentDidUpdate(){
+    //this.readFavorites();
+  }
+
+  async readFavorites() {
+    let api_key = 'Tpk_77a598a1fa804de592413ba39f6b137a';
+    console.log('reading favorites from DB');
+    let uid = firebase.auth().currentUser.uid;
+    let favoriteRef = database().ref(`${uid}/favorites/`);
+    let favorites = [];
+     await favoriteRef.once('value', (snapshot) => {
+      snapshot.forEach(function(childSnapshot) {
+        console.log(childSnapshot.key);
+        console.log(childSnapshot.val());
+        if(childSnapshot.val() === true) {
+          favorites.push(childSnapshot.key);
+        }
+      });
+    });
+
+    let companySymbolsAPI = favorites.join(',').toLowerCase();
+      const apiFetchURL = `https://sandbox.iexapis.com/stable/stock/market/batch?&types=quote&symbols=${companySymbolsAPI}&token=${api_key}`;
+      let companyNamesAPI = [];
+      try {
+        let response = await fetch(apiFetchURL);
+        let responseJson = await response.json();
+
+        const quotes = Object.values(responseJson).map((stock) => stock.quote);
+        const companyNames = quotes.map((quote) => quote.companyName);
+        companyNamesAPI = companyNames;
+      } catch (error) {
+        console.error(error);
+      }
+      console.log("FAVORITES: " + favorites);
+      console.log("FAVORITE NAMES: " + companyNamesAPI);
+      this.setState({
+        favoritedCompanies: {
+          companyNames: companyNamesAPI,
+          companySymbols: favorites,
+        },
+      });
```

Comment:
I would move this inside the try catch that way we don't set anything if an error happens during fetching.